ADAL/src/broker/ios/ADBrokerKeyHelper.m

```diff
@@ -287,4 +291,52 @@ + (void)setSymmetricKey:(NSString *)base64Key
     s_symmetricKeyOverride = base64Key ? [NSString adBase64UrlDecodeData:base64Key] : nil;
 }
 
++ (NSDictionary *) decryptBrokerResponse:(NSDictionary *) response correlationId:(NSUUID *) correlationId error:(ADAuthenticationError * __autoreleasing *)error
+{
+    NSString* hash = [response valueForKey:BROKER_HASH_KEY];
+    if (!hash)
+    {
+        AUTH_ERROR(AD_ERROR_TOKENBROKER_HASH_MISSING, @"Key hash is missing from the broker response", correlationId);
+        return nil;
+    }
+
+    NSString* encryptedBase64Response = [response valueForKey:BROKER_RESPONSE_KEY];
+    NSString* msgVer = [response valueForKey:BROKER_MESSAGE_VERSION];
+    NSInteger protocolVersion = 1;
+    if (msgVer)
+    {
+        protocolVersion = [msgVer integerValue];
+    }
+
+    //decrypt response first
+    ADBrokerKeyHelper* brokerHelper = [[ADBrokerKeyHelper alloc] init];
+    ADAuthenticationError* decryptionError = nil;
+    NSData *encryptedResponse = [NSString adBase64UrlDecodeData:encryptedBase64Response ];
+    NSData* decrypted = [brokerHelper decryptBrokerResponse:encryptedResponse
+                                                    version:protocolVersion
+                                                      error:&decryptionError];
+
+    if (!decrypted)
+    {
+        AUTH_ERROR_UNDERLYING(AD_ERROR_TOKENBROKER_DECRYPTION_FAILED, @"Failed to decrypt broker message", decryptionError, correlationId)
+        return nil;
+    }
+
+
+    NSString* decryptedString = [[NSString alloc] initWithData:decrypted encoding:NSUTF8StringEncoding];
```

Comment:
nit: check for decrypted string (initWithData can return nil if encoding mismatches)